modules/social_features/social_group/modules/social_group_request/src/Form/GroupRequestMembershipRequestForm.php

```diff
@@ -99,56 +55,66 @@ public function getFormId() {
    * {@inheritdoc}
    */
   public function buildForm(array $form, FormStateInterface $form_state, GroupInterface $group = NULL) {
-    $this->group = $group;
+    if (!$group) {
+      return [];
+    }
 
-    $form['description'] = [
-      '#type' => 'html_tag',
-      '#tag' => 'p',
-      '#value' => $this->t('You can leave a message in your request. Only when your request is approved, you will receive a notification via email and notification center.'),
-    ];
+    /** @var \Drupal\group\Entity\Storage\GroupRelationshipTypeStorageInterface $storage */
+    $storage = $this->entityTypeManager->getStorage('group_content_type');
+    $group_type_id = (string) $group->getGroupType()->id();
+    $relation_type_id = $storage->getRelationshipTypeId($group_type_id, 'group_membership_request');
+    $group_content = $this->entityTypeManager->getStorage('group_content')->create([
+      'type' => $relation_type_id,
+      'gid' => $group->id(),
+      'entity_id' => $this->currentUser()->id(),
+      'grequest_status' => GroupMembershipRequest::REQUEST_PENDING,
+    ]);
+    $this->setEntity($group_content);
 
-    $form['message'] = [
-      '#type' => 'textarea',
-      '#title' => $this->t('Message'),
-    ];
+    $form = parent::buildForm($form, $form_state);
 
-    $form['actions']['submit'] = [
-      '#type' => 'submit',
-      '#value' => $this->t('Send request'),
-      '#button_type' => 'primary',
-    ];
+    if (isset($form['field_grequest_message'])) {
+      $form['description'] = [
+        '#type' => 'html_tag',
+        '#tag' => 'p',
+        '#value' => $this->t('You can leave a message in your request. Only when your request is approved, you will receive a notification via email and notification center.'),
+        '#weight' => -10,
+      ];
+    }
```

Comment:
I think this is something we can instead move into `field.field.group_content.*.field_grequest_message` under the `description` string, right? Or does that not yield the same output?

Is there another way we could add that? The answer can also be yes but we do it in a follow-up. But I'd love to see if we can make this very specific class be mostly gone.